src/Toml/Bi/Combinators.hs

```diff
@@ -63,7 +65,7 @@ import Data.IntSet (IntSet)
 import Data.List.NonEmpty (NonEmpty (..), toList)
 import Data.Maybe (fromMaybe)
 import Data.Semigroup ((<>))
-import Data.Set (Set)
+import Data.Set (Set, fromList)
```

Comment:
Importing `fromList` seems redundant because you're using qualified `S.fromList`
```suggestion
import Data.Set (Set)
```